### PR TITLE
Searchbar added

### PR DESCRIPTION
**Pull request by komal-agarwal5**
## Related Issue
#365 
## Description
I have added search bar in leaderboard page to improve user experience

## Type of PR

- [ ] Bug fix
- [x] Feature enhancement
- [ ] Documentation update
- [ ] Other (specify): _______________

## Screenshots / videos (if applicable)
Screenshot 2024-07-10 193729
Screenshot 2024-07-10 193741

## Checklist:
- [x] I have performed a self-review of my code
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## Additional context:
Sir can you please check if it is working now as previously i was facing issues in loading the page on my localhost?

### CODE REVIEW

**Comment by ashutosh-rath02** [comment]
Thank you for submitting your pull request! 🙌 We'll review it as soon as possible. In the meantime, please ensure that your changes align with our [CONTRIBUTING.md](https://github.com/ashutosh-rath02/git-re/blob/main/CONTRIBUTING.md). If there are any specific instructions or feedback regarding your PR, we'll provide them here. Thanks again for your contribution! 😊

**Review by ashutosh-rath02 — CHANGES_REQUESTED** [review]
@komal-agarwal5 can you please resolve these?

image

File: app/leaderboard/page.tsx [hunk 1]
```diff
@@ -88,7 +142,7 @@ export default function Leaderboard() {
                   <TableCell className="font-medium flex items-center">
                     <Crown rank={overallRank} />
                     <Link
-                      href={`${process.env.NEXT_PUBLIC_URL}/resume/${user.username}`}
+                      href={${process.env.NEXT_PUBLIC_URL}/resume/${user.username}}
```
Comment (ashutosh-rath02) [inline app/leaderboard/page.tsx:145]: Why have you removed the ticks?

**Comment by komal-agarwal5** [comment]
Ok sir I will make the corrections soon

**Comment by komal-agarwal5** [comment]
https://github.com/user-attachments/assets/ce792001-c611-4121-8b62-304e121098b7

I have made the corrections, is it working now?
@ashutosh-rath02

**Review by ashutosh-rath02 — APPROVED** [review]
LGTM

**Review by ashutosh-rath02 — CHANGES_REQUESTED** [review]
@komal-agarwal5 If we search for a name in 2nd page we can't access it, you have to search from the db rather than from frontend directly.

**Comment by komal-agarwal5** [comment]
> @komal-agarwal5 If we search for a name in 2nd page we can't access it, you have to search from the db rather than from frontend directly.

Ok i will redo it accordingly

**Comment by ashutosh-rath02** [comment]
@komal-agarwal5 updates?

**Comment by komal-agarwal5** [comment]
> @komal-agarwal5 updates?

will complete by tomorrow eod

**Comment by komal-agarwal5** [comment]
> @komal-agarwal5 updates?

@ashutosh-rath02 

https://github.com/user-attachments/assets/3a3a1720-7c2b-4dc3-965e-01a0af108b26

This is the progress till now
I have integrated search from database instead of frontend and search function is working on same page as well as different page as i checked by adding 50 dummy datas
But the problem I am facing is when search is successful the user's rank is showing as 1 irrespective of its original rank
I have been trying to fix this but coudnt till now
So i request you to give me some more time to complete this

I have pushed the changes made till now ,you can check out if you want

**Comment by ashutosh-rath02** [comment]
Ok maybe @dhairyathedev can help.

**Comment by dhairyathedev** [comment]
> Ok maybe @dhairyathedev can help.

Okay, I will look into it.

**Comment by dhairyathedev** [comment]
@ashutosh-rath02 you can merge this PR. I will open another PR to fix this issue.

**Comment by ashutosh-rath02** [comment]
@all-contributor please add @komal-agarwal5 for code

**Review by ashutosh-rath02 — COMMENTED** [review]
LGTM

**Pull request merged**